src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeOptionalFieldsNode.cs

```diff
@@ -53,10 +57,31 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             if (!relocsOnly)
             {
                 _owner.ComputeOptionalEETypeFields(factory, relocsOnly: false);
-                objData.EmitBytes(_owner.GetOptionalFieldsData(factory));
+                objData.EmitBytes(_owner.GetOptionalFieldsData());
             }
             
             return objData.ToObjectData();
         }
+
+        protected override int ClassCode => (int)ObjectNodeOrder.EETypeOptionalFieldsNode;
+
+        protected override int CompareToImpl(ObjectNode other, TypeSystemComparer comparer, NameMangler nameMangler)
+        {
+            var compare = comparer.Compare(_owner.Type, ((EETypeOptionalFieldsNode)other)._owner.Type);
```

Comment:
This node is one of the reasons I marked the PR as [WIP] :)  I thought I could get away with one ClassCode for EETypes - this code is the downside to that approach. The presence of multiple EETypes per TypeDesc (a necessary and constructed existing in the same graph) means we need a ClassCode per EEType override though so I'll just pass along responsibility here.